client/client_test.go

```diff
@@ -29,6 +30,38 @@ import (
 	"time"
 )
 
+func Test_NewClient_ShouldSendAuthorizationHeaderWhenRetrievingApiLinks(t *testing.T) {
+	// given the expected environment variable exists
+	defer restoreGetEnvFunc()
+	defer clearEnv()
+	initTestEnv()
+	setEnv(config.FlyteJWTEnvName, "a.jwt.token")
+
+	// and we have a running server set to respond with flyte api links
+	ts, rec := mockServerWithRecorder(http.StatusCreated, flyteApiLinksResponse)
+	defer ts.Close()
+
+	// when we create a new client
+	baseUrl, _ := url.Parse(ts.URL)
+	NewClient(baseUrl, 10*time.Second)
+
+	// then
+	assert.Equal(t, "Bearer a.jwt.token", rec.reqs[0].Header.Get("Authorization"))
```

Comment:
Similar comment across the board: reqs[0] can be nil so this test would panic in that case instead of failing on assertion. Not important tho

Reply:
You're right - added an extra check